lib/src/model/product_type_filter.dart

```diff
@@ -0,0 +1,15 @@
+import 'off_tagged.dart';
+import 'product_type.dart';
+
+/// Filter on product type for API get product queries.
+class ProductTypeFilter implements OffTagged {
+  const ProductTypeFilter._(this.offTag);
+
+  ProductTypeFilter(final ProductType productType)
+      : offTag = productType.offTag;
+
+  static const all = ProductTypeFilter._('all');
```

Comment:
I think you missed the type here

Reply:
Indeed. Strange the lint check didn't send a warning.
I'll fix that.